db.js

```diff
@@ -116,6 +117,31 @@ if (couchUrl) {
   };
 
   module.exports.sanitizeResponse = sanitizeResponse;
+
+  const runBatch = (query, iteratee, batchSize, skip, callback) => {
+    query(skip, (err, response) => {
+      if (err) {
+        return callback(err);
+      }
+      console.log(`        Processing ${skip} to ${skip + batchSize} docs of ${response.total_rows} total`);
+      iteratee(response.rows, err => {
+        const keepGoing = response.total_rows > (skip + batchSize);
+        callback(err, keepGoing);
+      });
+    });
+  };
+
+  module.exports.batch = (query, iteratee, batchSize, callback) => {
```

Comment:
Yes probably but it's complicated so would need extensive testing. I've raised a separate issue to investigate this: medic/medic-webapp#3553